cli/tests/pcluster/config/test_update_policy.py

```diff
@@ -911,315 +911,607 @@ def test_compute_fleet_stop_fail_reason_and_actions_needed(
             None,
             False,
-            "All login and compute nodes must be stopped",
-            "Stop the compute fleet with the pcluster update-compute-fleet command. "
+            "All login nodes must be stopped.",
             "Stop the login nodes by setting Count parameter to 0 "
             "and update the cluster with the pcluster update-cluster command",
             "slurm",
-            id="running fleet and Efs change with no update strategy set",
+            id="Compute running, login running, add external FSxLustre, update strategy not set: rejected",
+        ),
+        pytest.param(
+            False,
+            True,
+            "SharedStorage",
+            [],
+            external_fsx_lustre,
+            None,
+            None,
+            False,
+            "All login nodes must be stopped.",
+            "Stop the login nodes by setting Count parameter to 0 "
+            "and update the cluster with the pcluster update-cluster command",
+            "slurm",
+            id="Compute running, login running, remove external FSxLustre, update strategy not set: rejected",
+        ),
+        pytest.param(
+            True,
+            True,
+            "SharedStorage",
+            [],
+            None,
+            external_fsx_lustre,
+            None,
+            False,
+            "All login nodes must be stopped.",
+            "Stop the login nodes by setting Count parameter to 0 "
+            "and update the cluster with the pcluster update-cluster command",
+            "slurm",
+            id="Compute stopped, login running, add external FSxLustre, update strategy not set: rejected",
+        ),
+        pytest.param(
+            True,
+            True,
+            "SharedStorage",
+            [],
+            external_fsx_lustre,
+            None,
+            None,
+            False,
+            "All login nodes must be stopped.",
+            "Stop the login nodes by setting Count parameter to 0 "
+            "and update the cluster with the pcluster update-cluster command",
+            "slurm",
+            id="Compute stopped, login running, remove external FSxLustre, update strategy not set: rejected",
+        ),
+        pytest.param(
+            False,
+            False,
+            "SharedStorage",
+            [],
+            None,
+            external_fsx_lustre,
+            None,
+            True,
+            None,
+            None,
+            "slurm",
+            id="Compute running, login stopped, add external FSxLustre, update strategy not set: accepted",
+        ),
+        pytest.param(
+            False,
+            False,
+            "SharedStorage",
+            [],
+            external_fsx_lustre,
+            None,
+            None,
+            True,
+            None,
+            None,
+            "slurm",
+            id="Compute running, login stopped, remove external FSxLustre, update strategy not set: accepted",
```

Comment:
We could move these shared storage tests into a separate `test_update_policy_shared_storage` file.
Great coverage!

Reply:
I will move in a dedicated PR. I prefer the refactoring to be decoupled from core PRs